File: CPP/hashmap/2225. Find Players With Zero or One Losses.cpp
class Solution {
public:
    vector<vector<int>> findWinners(vector<vector<int>>& matches) {
        unordered_map<int, int> loss_count;
        for (vector<int> match: matches){
            int winner = match[0], loser = match[1];
            if (loss_count.find(winner) == loss_count.end()){
                loss_count[winner] = 0;
            }
            loss_count[loser]++;
        }
        vector<vector<int>> res(2,vector<int>());
        for (auto [player, loss]: loss_count){
            if (loss == 0) res[0].push_back(player);
            else if (loss == 1) res[1].push_back(player);
        }
        sort(res[0].begin(), res[0].end());
        sort(res[1].begin(), res[1].end());
        return res;
        // unordered_map<int, int> loss_count;
        // set<int> players;
        // for (vector<int> match: matches){
        //     players.insert(match[0]);
        //     players.insert(match[1]);
        //     loss_count[match[1]]++;
        // }
        // vector<vector<int>> res;
        // vector<int> one_loss, zero_loss;
        // for (int p: players){
        //     if (loss_count[p] == 0) zero_loss.push_back(p);
        //     else if (loss_count[p] == 1) one_loss.push_back(p);
        // }
        // res.push_back(zero_loss);
        // res.push_back(one_loss);
        // return res;
    }
};

File: Python/Hashmap/2225. Find Players With Zero or One Losses.py
from typing import List

class Solution:
    def findWinners(self, matches: List[List[int]]) -> List[List[int]]:
        LoseCount = {}
        for match in matches:
            if match[0] not in LoseCount:
                LoseCount[match[0]] = 0
            if match[1] not in LoseCount:
                LoseCount[match[1]] = 1
            else:
                LoseCount[match[1]] += 1

        res = [[],[]]
        for player in LoseCount:
            if LoseCount[player] == 0:
                res[0].append(player)
            elif LoseCount[player] == 1:
                res[1].append(player)
        res[0] = sorted(res[0])
        res[1] = sorted(res[1])
        return res


if __name__ == "__main__":
    matches = [[1,3],[2,3],[3,6],[5,6],[5,7],[4,5],[4,8],[4,9],[10,4],[10,9]]
    sol = Solution()
    print(sol.findWinners(matches))